lib/radiator_web/resolvers/directory.ex

```diff
@@ -69,6 +69,22 @@ defmodule RadiatorWeb.Resolvers.Directory do
       nil ->
         {:error, "Podcast ID #{id} not found"}
 
+      %Podcast{slug: nil} = podcast ->
```

Comment:
My main concern with your current implementation is here. You implemented the slugging in the GraphQL resolver, so it relies on GraphQL being used -- which is an abstraction layer too high. It should apply when the main elixir API, `Directory.update_podcast/2`, is called. I suggest doing it inside the podcast changeset. 

Maybe https://github.com/sobolevn/ecto_autoslug_field is worth a closer look. I don't know if it's flexible enough for our use case but at least the slugging takes place in the right spot.

Reply:
Ok, putting it into `Radiator.Directory` was my initial thought. But when I was looking for `publish_podcast/3` and `depublish_podcast/3` I found them in `RadiatorWeb.Resolvers.Directory`. Should I move these two functions as well, so they are not only available in/for GraphQL?

Reply:
Right, that would be good!